src/validation/version.ts

```diff
@@ -0,0 +1,11 @@
+export function isValidWeaviateVersion(version: string) {
+    if (typeof version === 'string') {
+        const versionNumbers = version.split('.');
+        if (versionNumbers.length >= 2) {
+          const major = parseInt(versionNumbers[0], 10);
+          const minor = parseInt(versionNumbers[1], 10);
+          return (major <= 1 && minor < 16);
+        }
+    }
+    return false;
+}
```

Comment:
The boolean logic here is a little confusing since the function is named `isValidWeaviateVersion` yet it returns a falsey value on successful completion and truthy value on error/issue. With this logic, I would expect the function to be called `isNotValidWeaviateVersion`.

I think it would be better if this name was maintained but the internal logic was flipped so that an invalid version returns a falsey value.